.github/workflows/deploy-next-js-dev.yml

```diff
@@ -0,0 +1,111 @@
+      image: ${{ steps.docker-build-push.outputs.image }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ inputs.NODE_VERSION }}
+          cache: 'npm'
+      - name: next.js cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.npm
+            ${{ github.workspace }}/.next/cache
+          key: ${{ runner.os }}-nextjs-${{ hashFiles('**/package-lock.json') }}-${{ hashFiles('**/*.js', '**/*.jsx', '**/*.ts', '**/*.tsx') }}
+          restore-keys: |
+            ${{ runner.os }}-nextjs-${{ hashFiles('**/package-lock.json') }}-
+      - name: Install npm dependencies
+        env:
+          NPM_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: npm ci
+      - name: Pre-deploy
+        uses: navikt/pam-deploy/actions/pre-deploy@v7
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PROJECT_ID: ${{ vars.NAIS_MANAGEMENT_PROJECT_ID }}
+          TEAM: "teampam"
+          DRAFTS_MAX: "10"
+      - name: Run lint
+        run: npm run lint
+      - name: Run prettier
+        run: npm run prettier
+      - name: Build application
+        run: npm run build
+        env:
+          SENTRY_RELEASE: ${{ env.VERSION_TAG }}
+          SENTRY_AUTH_TOKEN: ${{ secrets.SENTRY_AUTH_TOKEN }}
+      - name: Upload static files to CDN
+        uses: nais/deploy/actions/cdn-upload/v2@master
+        with:
+          team: teampam
+          source: ./.next/static
+          destination: /${{github.event.repository.name}}/_next
+          project_id: ${{ vars.NAIS_MANAGEMENT_PROJECT_ID }}
+          identity_provider: ${{ secrets.NAIS_WORKLOAD_IDENTITY_PROVIDER }}
+      - name: docker-build-push
+        uses: nais/docker-build-push@v0
+        id: docker-build-push
+        with:
+          team: "teampam"
+          tag: ${{ env.VERSION_TAG }}
+          identity_provider: ${{ secrets.NAIS_WORKLOAD_IDENTITY_PROVIDER }}
+          project_id: ${{ vars.NAIS_MANAGEMENT_PROJECT_ID }}
+      - name: Deploy to ${{ inputs.CLUSTER }}
+        uses: nais/deploy/actions/deploy@v2
+        env:
+          CLUSTER: "dev-gcp"
+          RESOURCE: ".nais/nais.yml"
+          VARS: ${{ inputs.NAIS_VARS }}
+      - name: Post-deploy
+        uses: navikt/pam-deploy/actions/post-deploy@v7
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Draft release
+        if: ${{ !inputs.SKIP_DRAFT_RELEASE }}
+        uses: ncipollo/release-action@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag: ${{ env.VERSION_TAG }}
+          name: ${{ env.DRAFT_NAME }}
+          body: ${{ env.CHANGE_LOG }}
+          draft: true
+          prerelease: false
+
+  CodeQL:
+    uses: navikt/pam-deploy/.github/workflows/codeql.yml@next-js-deploy
+    with:
+      LANGUAGE: "node"
+      BUILD_CACHE: "npm"
```

Comment:
Funker denne nå for next-js apper?

Reply:
Ja, jag tror det. Den kör iaf och analyserar Javascript-filerna :thinking: